sheets/plugins.py

```diff
@@ -0,0 +1,93 @@
+from django.core.exceptions import ObjectDoesNotExist
+from mitol.google_sheets_refunds.hooks import hookimpl, RefundResult
+from mitol.google_sheets_refunds.utils import RefundRequestRow
+from mitol.google_sheets.utils import ResultType
+
+from courses.api import (
+    is_program_text_id,
+    deactivate_program_enrollment,
+    deactivate_run_enrollment,
+)
+from courses.constants import ENROLL_CHANGE_STATUS_REFUNDED
+from courses.models import ProgramEnrollment, CourseRunEnrollment
+from ecommerce.models import Order
+from users.models import User
+
+
+class RefundPlugin:
+    @staticmethod
+    def get_order_objects(refund_req_row):
+        """
+        Fetches all of the database objects relevant to this refund request
+
+        Args:
+            refund_req_row (RefundRequestRow): An object representing a row in the spreadsheet
+
+        Returns:
+            (Order, ProgramEnrollment or CourseRunEnrollment): The order and enrollment associated
+                with this refund request.
+        """
+        user = User.objects.get(email__iexact=refund_req_row.learner_email)
+        order = Order.objects.get(id=refund_req_row.order_id, purchaser=user)
+        if is_program_text_id(refund_req_row.product_id):
+            enrollment = ProgramEnrollment.all_objects.get(
+                order=order, program__readable_id=refund_req_row.product_id
+            )
+        else:
+            enrollment = CourseRunEnrollment.all_objects.get(
+                order=order, run__courseware_id=refund_req_row.product_id
+            )
+        return order, enrollment
+
+    @staticmethod
+    def reverse_order_and_enrollments(order, enrollment):
+        """
+        Sets the state of the given order and enrollment(s) to reflect that they have
+        been refunded and are no longer active
+
+        Args:
+            order (Order):
+            enrollment (ProgramEnrollment or CourseRunEnrollment):
+        """
+        if isinstance(enrollment, ProgramEnrollment):
+            deactivated_enrollment, _ = deactivate_program_enrollment(
+                enrollment, change_status=ENROLL_CHANGE_STATUS_REFUNDED
+            )
+        else:
+            deactivated_enrollment = deactivate_run_enrollment(
+                enrollment, change_status=ENROLL_CHANGE_STATUS_REFUNDED
+            )
```

Comment:
@rhysyngsun I don't have much context on sheets, And since you are reviewing, shouldn't we call [refund_order](https://github.com/mitodl/mitxonline/blob/main/ecommerce/api.py#L180) from ecommerce api, I believe that will call CyberSource and handle the refunding state of the Order in the database and return a bool value for success/failed refuned.

Reply:
@arslanashraf7 thanks, great catch, @annagav I think this means at least for now that this entire method would be replace by that call.